### PR TITLE
Updated build system

### PR DESCRIPTION
**Pull request by tnagorra**
Depends on https://github.com/the-deep/deeper/pull/965
Depends on https://github.com/the-deep/server/pull/890

### Update on deeper
- Pull latest changes
- Add these to .env on deeper
```bash
GRAPHQL_CODEGEN_ENDPOINT=http://localhost:8001/graphql
REACT_APP_GRAPHQL_ENDPOINT=http://localhost:8001/graphql
```
- Please update port 8000 to 8001 for all other configurations in .env
```bash
# Add / replace these keys
REACT_APP_API_END=localhost:8001
REACT_APP_ADMIN_END=localhost:8001
REACT_APP_API_HTTPS=http
REACT_APP_DEEP_ENVIRONMENT=development-nav
```

### Run proxy server to connect with other development server
- Clone https://github.com/toggle-corp/simple-proxy
- Add these to .env on simple-proxy
```bash
SOURCE_DOMAIN=nav-local.togglecorp.com
SOURCE_PROTOCOL=https
```
- Run `docker-compose up`

### Using client
- Build docker using
```bash
docker-compose build
```
- We don't have a login page yet
- Login using admin panel
- Oh no! you don't have username and password. Please contact @thenav56 jee!
- Run using `docker-compose up client`
- Don't forget to generate introspection schema

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,14 +11,6 @@ jobs:
     name: 🚴 Build + Lint + Test 🚴  # Match the name below (8398a7/action-slack).
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
```
Comment (thenav56) [inline .github/workflows/ci.yml:14]: This is needed to fetch the-deep/client

**Review by frozenhelium — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tnagorra) [reply .github/workflows/ci.yml:14]: Thanks!

**Review by AdityaKhatri — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -1,216 +1,127 @@
 {
-    "name": "deep",
+    "name": "base-react-app",
```
Comment (AdityaKhatri) [inline package.json]: Rename to DEEP

File: package.json [hunk 3]
```diff
@@ -1,216 +1,127 @@
...
-    "description": "Deep",
-    "main": "index.js",
-    "keywords": [],
-    "author": "",
-    "license": "ISC",
-    "sideEffects": [
-        "@the-deep/deep-ui/build/index.css",
-        "./src/Multiplexer/styles.scss"
-    ],
+    "description": "Base React Application",
```
Comment (AdityaKhatri) [inline package.json]: Rename to DEEP

File: package.json [hunk 4]
```diff
@@ -1,216 +1,127 @@
...
-        },
-        "transformIgnorePatterns": [
-            "[/\\\\]node_modules[/\\\\].+\\.(js|jsx)$"
-        ]
+    "sideEffects": [
+        "@the-deep/deep-ui/build/index.css"
+    ],
+    "repository": {
+        "type": "git",
+        "url": "git+https://github.com/togglecorp/base-react-app.git"
```
Comment (AdityaKhatri) [inline package.json]: Rename to DEEP

File: package.json [hunk 5]
```diff
@@ -1,216 +1,127 @@
...
-        "webpack": "^4.43.0",
-        "webpack-cli": "^3.3.12",
-        "webpack-dev-server": "^3.11.0",
-        "webpack-merge": "^5.0.8",
-        "webpack-pwa-manifest": "^4.2.0"
+    "keywords": [],
+    "author": "",
+    "license": "ISC",
+    "bugs": {
+        "url": "https://github.com/togglecorp/base-react-app/issues"
```
Comment (AdityaKhatri) [inline package.json]: Rename to DEEP

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]
Rename base-react-app to DEEP and properly maintain brand name.
Also, please rebase with develop-deep and send a PR.

File: .gitignore [hunk 6]
```diff
@@ -2,8 +2,6 @@
 
 # dependencies
 /node_modules
-vendor/
```
Comment (AdityaKhatri) [inline .gitignore:5]: Add this back for now

File: package.json (hunk 5 above)
Comment (tnagorra) [reply package.json]: okay sir!

**Review by frozenhelium — APPROVED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**